src/Model/xrRtstruct.py

```diff
@@ -0,0 +1,165 @@
+import pydicom
+from pydicom import dcmread
+from pydicom.dataset import FileMetaDataset, validate_file_meta
+from pydicom.uid import generate_uid, ImplicitVRLittleEndian
+from pydicom.tag import Tag
+from pathlib import Path
+from copy import deepcopy
+import datetime
+
+#AttributeError: 'Dataset' object has no attribute 'FrameOfReferenceUID'
+#removed 2 lines relating to this error, might need to be looked at if this is required
```

Comment:
For the RT SS to be DICOM conformant (and can have an expectation of being usable by anything else), it will need to have *a* FrameOfReferenceUID.  Given that one is trying to tie the RT SS to the CR, one could generate a new UID for the FoR, insert that in to the RT SS, and update the CR object by adding the FoR UID to that.  Given that you know you will need to do this up front, it would be better to update the CR first (with a generated FoR UID), and update the SOP Instance UID of the CR (because it's a new object and not interchangeable with the original CR that lacks the FoR UID), and then use that FoR UID in the RT SS.  Having a DICOM object with a public element it wouldn't normally have tends to be reasonably interoperable.  Having a DICOM object without a public element that it is required to have is problematic.